python/python/lance/dataset.py

```diff
@@ -612,6 +612,38 @@ def delete(self, predicate: Union[str, pa.compute.Expression]):
             predicate = str(predicate)
         self._ds.delete(predicate)
 
+    def update(
+        self,
+        updates: Dict[str, str],
+        where: Optional[str] = None,
+    ):
+        """
+        Update column values for rows matching where.
+
+        Parameters
+        ----------
+        updates : dict of str to str
+            A mapping of column names to a SQL expression.
+        where : str, optional
+            A SQL predicate indicating which rows should be updates.
+
+        Examples
+        --------
+        >>> import lance
+        >>> import pyarrow as pa
+        >>> table = pa.table({"a": [1, 2, 3], "b": ["a", "b", "c"]})
+        >>> dataset = lance.write_dataset(table, "example")
+        >>> dataset.update(dict(a = 'a + 2'), where="b > 'a'")
```

Comment:
```suggestion
        >>> dataset.update(dict(a = 'a + 2'), where="b != 'a'")
```
Minor nit:  maybe use `b != 'a'` here.  String comparison is slightly more niche.